File: resources/expected/ros2/datatype-examples/lax/src/datatypes_system_cpp_pkg/src/base_code/enum_converter.cpp
#include "datatypes_system_cpp_pkg/base_headers/enum_converter.hpp"

const char* enumToString(datatypes_system_cpp_pkg_interfaces::msg::MyEnum* value) {
    switch (value->my_enum) {
        case datatypes_system_cpp_pkg_interfaces::msg::MyEnum::MY_ENUM_ON:
            return "MyEnum On";
        case datatypes_system_cpp_pkg_interfaces::msg::MyEnum::MY_ENUM_OFF:
            return "MyEnum Off";
        default:
            return "Unknown value for MyEnum";
    }
}

File: resources/expected/ros2/python-datatype-examples/lax/src/datatypes_system_py_pkg/datatypes_system_py_pkg/base_code/enum_converter.py
#!/usr/bin/env python3
from datatypes_system_py_pkg_interfaces.msg import MyEnum

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

def enumToString(value):
    typedValue = MyEnum()
    typedValue.data = value
    match (typedValue.my_enum):
        case MyEnum.MY_ENUM_ON:
            return "MyEnum On"
        case MyEnum.MY_ENUM_OFF:
            return "MyEnum Off"
        case default:
            return "Unknown value for MyEnum"

